src/operator/tensor/dot-inl.h

```diff
@@ -1207,6 +1207,14 @@ inline bool DotShape(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(out_attrs->size(), 1U);
   mxnet::TShape& lshape = (*in_attrs)[0];
   mxnet::TShape& rshape = (*in_attrs)[1];
+  // check if lhs ndim is larger than 1 and last dim is known
+  if (lshape.ndim() < 1 || !dim_size_is_known(lshape, lshape.ndim() - 1)) {
```

Comment:
`ndim=0` is a valid case representing scalar tensors. This line should be replaced by the following.
```cpp
if (!ndim_is_known(lshape) || !ndim_is_known(rshape)) return false;
CHECK_GT(lshape.ndim(), 0) << "scalar tensor is not supported by this operator.";
CHECK_GT(rshape.ndim(), 0) << "scalar tensor is not supported by this operator.";
```